src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -91,11 +92,19 @@ public void CloneTo(X509Certificate2Collection collection)
                 return;
             }
 
+            var loadedCerts = new HashSet<X509Certificate2>();
+
             foreach (string filePath in Directory.EnumerateFiles(_storePath, PfxWildcard))
             {
                 try
                 {
-                    collection.Add(new X509Certificate2(filePath));
+                    var cert = new X509Certificate2(filePath);
+
+                    // If we haven't already loaded a cert .Equal to this one, copy it to the collection.
+                    if (loadedCerts.Add(cert))
```

Comment:
I know we discussed this briefly earlier, but... we're not concerned about the performance impact of this, such that we're getting a hashcode that's based on state that may not otherwise be loaded?